misk-hibernate/src/main/kotlin/misk/vitess/StartVitessService.kt

```diff
@@ -262,24 +276,30 @@ class DockerVitessCluster(
   }
 
   private fun waitUntilHealthy() {
-    retry(10, ExponentialBackoff(Duration.ofMillis(20), Duration.ofMillis(5000))) {
-      cluster.openVtgateConnection().use { c ->
-        try {
-          val result =
-              c.createStatement().executeQuery("SELECT 1 FROM dual").uniqueInt()
-          check(result == 1)
-        } catch (e: Exception) {
-          val message = e.message
-          if (message?.contains("table dual not found") == true) {
-            throw RuntimeException(
-                "Something is wrong with your vschema and unfortunately vtcombo does not " +
-                    "currently have good error reporting on this. Please inspect the logs or your " +
-                    "vschema to see if you can find the error.")
-          } else {
-            throw e
+    try {
+      retry(20, ExponentialBackoff(Duration.ofSeconds(1), Duration.ofSeconds(5))) {
+        cluster.openVtgateConnection().use { c ->
+          try {
+            val result =
+                c.createStatement().executeQuery("SELECT 1 FROM dual").uniqueInt()
+            check(result == 1)
+          } catch (e: Exception) {
+            val message = e.message
+            if (message?.contains("table dual not found") == true) {
+              throw DontRetryException(
+                  "Something is wrong with your vschema and unfortunately vtcombo does not " +
+                      "currently have good error reporting on this. Please inspect the logs or your " +
+                      "vschema to see if you can find the error.")
+            } else {
+              throw e
+            }
           }
         }
       }
+    } catch (e: DontRetryException) {
+      throw Exception(e.message)
+    } catch (e: Exception) {
+      throw Exception("Vitess cluster failed to start up in time", e)
```

Comment:
Why this rethrowing?

Reply:
We still want to see this error message in the test logs. We just need to enrich it with a better message.